File: client-clojure/src/sirqul_io_t_platform/specs/asset_short_response.clj
(ns sirqul-io-t-platform.specs.asset-short-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            )
  (:import (java.io File)))


(def asset-short-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :assetId) int?
   (ds/opt :fileName) string?
   (ds/opt :caption) string?
   (ds/opt :mediaType) string?
   (ds/opt :thumbnailURL) string?
   (ds/opt :coverURL) string?
   (ds/opt :fullURL) string?
   (ds/opt :assetVersion) string?
   (ds/opt :versionName) string?
   (ds/opt :versionCode) int?
   (ds/opt :returning) string?
   })

(def asset-short-response-spec
  (ds/spec
    {:name ::asset-short-response
     :spec asset-short-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/AssetShortResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import AssetShortResponse._

case class AssetShortResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
assetId: Option[Long],
fileName: Option[String],
caption: Option[String],
mediaType: Option[String],
thumbnailURL: Option[String],
coverURL: Option[String],
fullURL: Option[String],
assetVersion: Option[String],
versionName: Option[String],
versionCode: Option[Integer],
returning: Option[String])

object AssetShortResponse {
  import DateTimeCodecs._

  implicit val AssetShortResponseCodecJson: CodecJson[AssetShortResponse] = CodecJson.derive[AssetShortResponse]
  implicit val AssetShortResponseDecoder: EntityDecoder[AssetShortResponse] = jsonOf[AssetShortResponse]
  implicit val AssetShortResponseEncoder: EntityEncoder[AssetShortResponse] = jsonEncoderOf[AssetShortResponse]
}
